Add vglite examples as independent library or framework

NGExplorer opened the pull request:
This pull request aim to address issue https://github.com/zephyrproject-rtos/zephyr/issues/100436
This PR adds 6 test applications (cube, clock, vector, tiger, tile, toolkit, decompress_etc2)

zejiang0jason commented on boards/nxp/mimxrt1170_evk/nxp_gpu_interface.h:
```diff
@@ -0,0 +1,225 @@
+ */
+
+#define CACHEABLE_START   0x80000000U
+#define CACHEABLE_END     0x83FFFFFFU
```
not good to use fixed values.

zejiang0jason commented on west.yml:
```diff
@@ -390,6 +390,10 @@ manifest:
       revision: 9b07780aca6fb21f82a241ba386ad9b379809337
       path: modules/lib/zcbor
+    - name: gpu-vglite
+      url: https://github.com/nxp-mcuxpresso/gpu-vglite.git
```
maybe need to fork one repo in zephyr project.

zejiang0jason commented on kernel/CMakeLists.txt:
```diff
@@ -71,6 +71,7 @@ kernel_sources(
   float.c
   version.c
   )
+zephyr_library_sources_ifdef(CONFIG_TIMING_FUNCTIONS busy_wait.c)
```
seems this file is added here: https://github.com/zephyrproject-rtos/zephyr/blob/main/kernel/CMakeLists.txt#L64.

NGExplorer replied:
Incorporated suggestion.

NGExplorer commented on include/zephyr/arch/arm/cortex_m/scripts/linker.ld:
```diff
@@ -456,4 +456,12 @@ _flash_used = LOADADDR(.last_section) + SIZEOF(.last_section) - __rom_region_sta
+    KEEP(*(.vglite.noinit.mem_pool.*))
+    . = ALIGN(4);
+} > RAM
+
```
Incorporated suggestion.

NGExplorer commented on scripts/build/parse_syscalls.py:
```diff
@@ -72,7 +72,7 @@ def analyze_headers(include_dir, scan_dir, file_list):
     # If file_list does not exist, we emit all syscalls.
     if file_list:
-        with open(file_list, encoding="utf-8") as fp:
+        with open(file_list, encoding="utf-8", errors="ignore") as fp:
```
Incorporated suggestion.

NGExplorer commented on samples/subsys/gpu/vglite_examples/vglite_window/vglite_window.c:
```diff
@@ -0,0 +1,90 @@
+#include <zephyr/sys/util.h>
+#include <zephyr/kernel.h>
+
+__aligned(FRAME_BUFFER_ALIGN) static uint8_t s_frameBuffer[APP_BUFFER_COUNT][720 * 1280 * 2];
```
Incorporated suggestion.

NGExplorer commented on samples/subsys/gpu/vglite_examples/tile/prj.conf:
Incorporated suggestion.

NGExplorer commented on samples/subsys/gpu/vglite_examples/tile/prj.conf:
Incorporated suggestion.

NGExplorer commented on samples/subsys/gpu/vglite_examples/tile/CMakeLists.txt:
```diff
@@ -0,0 +1,51 @@
+zephyr_include_directories(${ZEPHYR_BASE}/include/zephyr)
+
+if(${CONFIG_SOC_PART_NUMBER_MIMXRT1176DVMAA})
+  zephyr_include_directories(${ZEPHYR_BASE}/boards/nxp/mimxrt1170_evk)
```
Incorporated suggestion.

NGExplorer commented on include/zephyr/arch/arm/cortex_m/scripts/linker.ld:
```diff
@@ -456,4 +456,12 @@ _flash_used = LOADADDR(.last_section) + SIZEOF(.last_section) - __rom_region_sta
     GROUP_END(RAMABLE_REGION)
 
+/* VG Lite heap in RAM, noinit */
+.vglite.noinit.mem_pool.* (NOLOAD) :
```
Incorporated suggestion.

kartben reviewed (changes requested):
Please contribute this as an external module https://docs.zephyrproject.org/latest/develop/manifest/index.html#west-external-projects. You can host the samples in that external module, and add basic instructions pointing to them in the Zephyr  documentation of this new external module.

What's more, there is no such thing as a "GPU" subsystem in Zephyr so adding GPU subsys samples would be a bit awkward.

HKPTech committed:
vglite-examples: Add vglite examples as external module

Following gpu-vglite tests for IMXRT595, IMXRT1170 and IMXRT700 SoCs.
  * clock
  * cube
  * decompress_etc2
  * tiger
  * tile
  * toolkit
  * vector

Incorporating gpu-vglite and gpu-vglite-examples as external modules.

Add external manifest documentation entries for gpu-vglite-examples
and gpu-vglite, describing their purpose and how they can be included
as external projects.

Signed-off-by: Harikrushna Parmar <harikrushna.parmar@nxp.com>

kartben reviewed (changes requested):
Commit message now needs to be updated
Also, please consolidate everything into a single documentation page, it doesn't really make sense to advertise examples independently from the rest?

Last but not least, the gpu-vglite repo is not a west module (https://docs.zephyrproject.org/latest/develop/modules.html#modules-vs-west-projects) so I am actually not sure what you are trying to accomplish with this pull request?

butok commented:
Just like other Zephyr external modules, for simplicity, is it possible to use one external module containing samples instead of two separate modules? Or at least to have one VGlite documentation page describing both?

butok commented on doc/develop/manifest/external/gpu-vglite-examples.rst:
```diff
@@ -0,0 +1,77 @@
+examples leverage:
+
+- Zephyr's display driver interface
+- VGLite GPU hardware abstraction
```
BTW:  For future, it can be a basis of a common Zephyr GPU abstraction layer/driver/API - Zephyr gap.
